kadena/core/src/types/header/chain.rs

```diff
@@ -236,18 +238,19 @@ impl KadenaHeaderRaw {
     /// # Returns
     ///
     /// The root hash of the header.
-    pub fn header_root(&self) -> Result<HashValue, CryptoError> {
-        let adjacent_hashes: Vec<[u8; 32]> = vec![
-            self.adjacents[6..6 + DIGEST_BYTES_LENGTH]
+    pub fn header_root(&self) -> Result<HashValue, TypesError> {
```

Comment:
```suggestion
    pub fn header_root(&self) -> Result<HashValue, CryptoError> {
```

Reply:
The tree logic further down in this PR would have to change, too when the degree changes. This should at least be documented in the code.